sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/RegexpExpressionsSuite.scala

```diff
@@ -74,6 +107,31 @@ class RegexpExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation("a\nb" like regEx, true, create_row("a%b"))
 
     checkEvaluation(Literal.create(null, StringType) like regEx, null, create_row("bc%"))
+
+    checkEvaluation("" like regEx, true, create_row(""))
+    checkEvaluation("a" like regEx, false, create_row(""))
+    checkEvaluation("" like regEx, false, create_row("a"))
+
+    checkEvaluation("""\\\\""" like regEx, true, create_row("""%\\%"""))
+    checkEvaluation("""%%""" like regEx, true, create_row("""%%"""))
+    checkEvaluation("""\__""" like regEx, true, create_row("""\\\__"""))
+    checkEvaluation("""\\\__""" like regEx, false, create_row("""%\\%\%"""))
+    checkEvaluation("""_\\\%""" like regEx, false, create_row("""%\\"""))
+
+    // scalastyle:off nonascii
+    checkEvaluation("a\u20ACa" like regEx, true, create_row("_\u20AC_"))
+    checkEvaluation("a€a" like regEx, true, create_row("_€_"))
+    checkEvaluation("a€a" like regEx, true, create_row("_\u20AC_"))
+    checkEvaluation("a\u20ACa" like regEx, true, create_row("_€_"))
+    // scalastyle:on nonascii
+
+    // TODO: should throw an exception?
```

Comment:
> The escape character must be a single character no more than two bytes in length. It can only appear in the pattern string if it is followed by itself, a percent sign, or an underscore.

does this mean that PostgreSQL's behaviour is wrong? Should spark throw an exception in point 2?